storey/flow.py

```diff
@@ -1552,7 +1594,124 @@ def _static_run(*args, **kwargs):
     return _sval._run(*args, **kwargs)
 
 
-class ParallelExecution(Flow):
+class Executor:
+    def __init__(
+        self,
+        max_processes: Optional[int] = None,
+        max_threads: Optional[int] = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        self.runnables: dict[str, ParallelExecutionRunnable] = {}
+        self._runnable_by_name = {}
+
+        self.max_processes = max_processes or os.cpu_count() or 16
+        self.max_threads = max_threads or 32
+        self.num_processes = 0
+        self.num_threads = 0
+
+        self._process_executor_by_runnable_name = {}
+        self._mp_context = None
+
+    def add_runnable(self, runnable: ParallelExecutionRunnable) -> None:
```

Comment:
Do we need to make it mutable? It can be mutable in mlrun and immutable at runtime (this is a runtime object).

Reply:
I think that in MLRUN I will add the runnable it only at runtime

Reply:
I'm pretty sure you don't want to add it at runtime (i.e. while already handling events). And `RunnableExecutor` should only exist at runtime, not when deploying the function SDK-side or server-side.